tests/foreman/ui/test_docker.py

```diff
@@ -310,6 +289,43 @@ def test_positive_update_upstream_name(self):
                     self.assertTrue(self.repository.validate_field(
                         repo_name, 'upstream', new_upstream_name))
 
+    @run_only_on('sat')
+    @tier1
+    def test_negative_update_upstream_name(self):
+        """Attempt to update upstream name for a Docker-type repository.
+
+        :id: 4722cfa1-33d0-41c4-8ed2-46da9b6d0cd1
+
+        :expectedresults: A repository is created with a Docker upstream
+            repository and that its upstream name can not be updated with
+            invalid values.
+
+        :CaseImportance: Critical
+        """
+        with Session(self) as session:
+            repo_name = gen_string('alphanumeric')
+            product = entities.Product(organization=self.organization).create()
+            _create_repository(
+                session,
+                org=self.organization.name,
+                name=repo_name,
+                product=product.name,
+            )
+            self.assertIsNotNone(self.repository.search(repo_name))
+            self.assertTrue(self.repository.validate_field(
+                repo_name, 'upstream', 'busybox'))
+            for new_upstream_name in invalid_docker_upstream_names():
+                with self.subTest(new_upstream_name):
+                    self.products.search_and_click(product.name)
```

Comment:
I feel like that `self.products.search_and_click(product.name)` can be put out of the loop, but not sure

Reply:
`self.repository.update` assume that user located product page, while `self.repository.update` goes inside repo. So we need `self.products.search_and_click(product.name)`